sample/.github/workflows/test.yml

```diff
@@ -1,18 +1,53 @@
-name: Test
+name: Test and code analysis
 on:
   pull_request:
-    types: [opened, synchronize, reopened]
+    types: [ opened, synchronize, reopened, ready_for_review ]
 
 jobs:
   test:
     name: Test
     runs-on: ubuntu-latest
     timeout-minutes: 30
     steps:
-      - name: Check out
+      - name: Set up JDK 17
```

Comment:
Optional: I think we can name it simply `Set up JDK`; if we ever need to update the version, we can only update it.